test/e2e/destroycluster/destroy_test.go

```diff
@@ -0,0 +1,100 @@
+package destroycluster
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/cache"
+	clientwatch "k8s.io/client-go/tools/watch"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	"github.com/openshift/hive/pkg/constants"
+	"github.com/openshift/hive/test/e2e/common"
+)
+
+const (
+	timeout = 30 * time.Minute
+)
+
+func TestDestroyCluster(t *testing.T) {
+	logger := log.WithField("test", "TestDestroyCluster")
+	logger.Info("Starting destroy cluster test")
+
+	cd := common.MustGetClusterDeployment()
+	logger = logger.WithField("cluster", fmt.Sprintf("%s/%s", cd.Namespace, cd.Name))
+	if !cd.Spec.Installed {
+		logger.Warn("Cluster deployment was not installed")
+	}
+
+	fail := failTestFunc(t, logger)
+
+	c := common.MustGetClient()
+	logger.Info("Deleting cluster deployment")
+	err := c.Delete(context.TODO(), cd)
+	if err != nil {
+		fail("Failed to delete cluster: %v", err)
+	}
+
+	logger.Info("Waiting for cluster deployment to go away")
+	waitForClusterDeploymentToGoAway(cd, fail)
+	logger.Info("Cluster deployment has been removed")
+
+	// Ensure that no dnszones owned by the clusterdeployment are left over
+	dnsZoneList := &hivev1.DNSZoneList{}
+	c.List(context.TODO(), dnsZoneList,
+		client.InNamespace(cd.Namespace),
+		client.MatchingLabels{
+			constants.DNSZoneTypeLabel:           constants.DNSZoneTypeChild,
+			constants.ClusterDeploymentNameLabel: cd.Name,
+		})
+	if len(dnsZoneList.Items) > 0 {
+		fail("At least one DNSZone still exists after clusterdeployment deletion")
+	}
+	logger.Info("No DNS Zones related to the cluster exist")
+}
+
+func waitForClusterDeploymentToGoAway(cd *hivev1.ClusterDeployment, fail func(string, ...interface{})) {
```

Comment:
Would it be possible to confirm we saw an uninstall job here, if we expected there to be one? InfraID does transfer to the ClusterDeployment even when the install is underway, so in theory if that is set, I think we *should* see an uninstall job at some point.

Reply:
Yup, will add code to monitor the uninstall job and output the log